solidity/contracts/ECDSABackportRewards.sol

```diff
@@ -0,0 +1,127 @@
+
+contract ECDSABackportRewards is Rewards {
+    // BondedECDSAKeepFactory deployment date, May-13-2020 interval started.
+    // https://etherscan.io/address/0x18758f16988E61Cd4B61E6B930694BD9fB07C22F
+    uint256 internal constant bondedECDSAKeepFactoryDeployment = 1589408351;
+
+    // We are going to have one interval, with a weight of 100%.
+    uint256[] internal backportECDSAIntervalWeight = [100];
+
+    // Interval is the difference in time of creation between older and newer
+    // versions of BondedECDSAKeepFactory.
+    // Older: https://etherscan.io/address/0x18758f16988E61Cd4B61E6B930694BD9fB07C22F
+    // Newer: https://etherscan.io/address/0xA7d9E842EFB252389d613dA88EDa3731512e40bD
+    uint256 internal constant backportECDSATermLength = 123 days; // 10678946 sec
+
+    // There were 41 Keeps created by BondedECDSAKeepFactory : 0x18758f16988E61Cd4B61E6B930694BD9fB07C22F
+    // The last Keep was opened on May-17-2020
+    // https://etherscan.io/address/0x45A3cACA2F2a78A53607618651C86111c9720AA5
+    uint256 internal constant numberOfCreatedECDSAKeeps = 41;
+
+    // We allocate ecdsa backport rewards to all 41 keeps.
+    uint256
+        internal constant minimumECDSAKeepsPerInterval = numberOfCreatedECDSAKeeps;
+
+    BondedECDSAKeepFactory factory;
+
+    constructor(address _token, address payable _factoryAddress)
+        public
+        Rewards(
+            _token,
+            bondedECDSAKeepFactoryDeployment,
+            backportECDSAIntervalWeight,
+            backportECDSATermLength,
+            minimumECDSAKeepsPerInterval
+        )
+    {
+        factory = BondedECDSAKeepFactory(_factoryAddress);
+    }
+
+    function _getKeepCount() internal view returns (uint256) {
+        // Between May 13 2020 - Sep 14 2020 there were 41 keeps opened.
+        return numberOfCreatedECDSAKeeps;
+    }
+
+    function _getKeepAtIndex(uint256 i) internal view returns (bytes32) {
+        return bytes32(i);
+    }
+
+    function _getCreationTime(bytes32) internal view returns (uint256) {
+        // Assign each keep to the starting timestamp of its interval.
+        return startOf(0);
+    }
+
+    function _isClosed(bytes32) internal view returns (bool) {
+        // All keeps created between May 13 2020 - Sep 14 2020 are considered closed.
+        return true;
+    }
+
+    function _isTerminated(bytes32 groupIndexBytes)
+        internal
+        view
+        returns (bool)
+    {
+        return false;
+    }
+
+    // A keep is recognized if its creation time is between May-13-2020 - Sep 14 2020
+    function _recognizedByFactory(bytes32 _keep) internal view returns (bool) {
+        uint256 keepCreatedTimestamp = factory.getKeepOpenedTimestamp(
+            toAddress(_keep)
+        );
+        if (keepCreatedTimestamp == 0) {
+            return false;
+        }
+        return
+            keepCreatedTimestamp <
+            bondedECDSAKeepFactoryDeployment.add(backportECDSATermLength);
```

Comment:
Do we need it? Given that the interval has 100% weight, it has already ended, and all keeps are considered happily closed, we should be good without this check. I think it is enough to check whether the creation date is non-zero.

Reply:
Yea, you're right, I think we can remove that additional check.